### ISSUE TITLE
Twig\Error\RuntimeError when core:loginpage_links are set

### ISSUE THREAD

**Issue by ghost** [simplesamlphp/simplesamlphp#1770]
**Describe the bug**
When specifying login links in an authentication source (via the config property `core:loginpage_links`) a twig runtime error is emitted: `Twig\Error\RuntimeError: Variable "pageid" does not exist.` The only two references to `pageid` are:
```
 $ grep -rFi pageid simplesamlphp/
simplesamlphp/locales/lt/LC_MESSAGES/messages.po:"Jei pageidaujate, kad administratorius su Jumis susisiektų, įveskite savo"
simplesamlphp/modules/core/templates/base.twig:    {% if name == pageid %}
```
The second and relevant one belongs to the loop over the data provided by
```
simplesamlphp/modules/core/src/Controller/Login.php:            $t->data['links'] = $source->getLoginLinks();
```
and pageid seems to only be used to toggle link active state:
```
    {% if links is defined %}                                                                                           
    {% for name, link in links %}                                                                                       
    {% if name == pageid %}                                                                                             
        <li class="ui-state-default ui-corner-top ui-tabs-selected ui-state-active">                                    
            <a href="#">{{ link.text|trans }}</a>                                                                       
        </li>                                                                                                           
    {% else %}                                                                                                          
        <li class="ui-state-default ui-corner-top">                                                                     
            <a href="{{ link.href }}">{{ link.text|trans }}</a>                                                         
        </li>                                                                                                           
    {% endif %}                                                                                                         
    {% endfor %}                                                                                                        
    {% endif %}
```

Not quite sure what to make out of this, either `pageid` could/should be `templateId` or only the else part should be there in the first place?


**To Reproduce**
Define an authentication source that sets `core:loginpage_links` like
```
<?php                                                                                                                   
declare(strict_types = 1);                                                                                              

namespace SimpleSAML\Module\pageiderror\Auth\Source;                                                                    

use SimpleSAML\Module\core\Auth\UserPassBase;                                                                           
                                                                                                                        
class PageIdErrorSource extends UserPassBase                                                                            
{                                                                                                                       
    public function __construct($info, &$config)                                                                        
    {                                                                                                                   
        $config['core:loginpage_links'] = [['href' => 'link', 'text' => 'link text']];                                  
                                                                                                                        
        parent::__construct($info, $config);                                                                            
    }                                                                                                                   
                                                                                                                        
    protected function login(string $username, string $password): array                                                 
    {                                                                                                                   
        return ['attribute' => 'Welcome'];                                                                              
    }                                                                                                                   
}    
```

**Expected behavior**
The links to be rendered on the login page.

**Screenshots or logs**
```
SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Backtrace:
2 .../simplesamlphp/simplesamlphp/public/_include.php:28 (SimpleSAML_exception_handler)
1 .../symfony/error-handler/ErrorHandler.php:607 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
0 [builtin] (N/A)
Caused by: SimpleSAML\Error\Exception: Variable "pageid" does not exist in core:loginuserpass.twig
Backtrace:
2 .../simplesamlphp/simplesamlphp/src/SimpleSAML/XHTML/Template.php:562 (SimpleSAML\XHTML\Template::getContents)
1 .../simplesamlphp/simplesamlphp/src/SimpleSAML/XHTML/Template.php:577 (SimpleSAML\XHTML\Template::send)
0 .../simplesamlphp/simplesamlphp/public/module.php:14 (N/A)
Caused by: Twig\Error\RuntimeError: Variable "pageid" does not exist.
Backtrace:
17 .../simplesamlphp/simplesamlphp/modules/core/templates/base.twig:10 (__TwigTemplate_6f15f4b9570b86b4c8f495ba4c31f3bae10954afcd90ecac0f569d29a775f78b::{closure})
16 .../twig/twig/src/Environment.php(361) : eval()'d code:65 (__TwigTemplate_6f15f4b9570b86b4c8f495ba4c31f3bae10954afcd90ecac0f569d29a775f78b::block_contentwrapper)
15 .../twig/twig/src/Template.php:171 (Twig\Template::displayBlock)
14 .../twig/twig/src/Environment.php(361) : eval()'d code:96 (__TwigTemplate_76053e00872c1a3aa3d43e53d596f44f1fa0ae64f2217eea8ae3827279361d93::doDisplay)
13 .../twig/twig/src/Template.php:394 (Twig\Template::displayWithErrorHandling)
12 .../twig/twig/src/Template.php:367 (Twig\Template::display)
11 .../twig/twig/src/Environment.php(361) : eval()'d code:42 (__TwigTemplate_6f15f4b9570b86b4c8f495ba4c31f3bae10954afcd90ecac0f569d29a775f78b::doDisplay)
10 .../twig/twig/src/Template.php:394 (Twig\Template::displayWithErrorHandling)
9 .../twig/twig/src/Template.php:367 (Twig\Template::display)
8 .../twig/twig/src/Environment.php(361) : eval()'d code:46 (__TwigTemplate_d74faf58cdbff543d0b213a8af0c246c1f8df6f17d6e9c37ae49237b343df563::doDisplay)
7 .../twig/twig/src/Template.php:394 (Twig\Template::displayWithErrorHandling)
6 .../twig/twig/src/Template.php:367 (Twig\Template::display)
5 .../twig/twig/src/Template.php:379 (Twig\Template::render)
4 .../twig/twig/src/TemplateWrapper.php:40 (Twig\TemplateWrapper::render)
3 .../twig/twig/src/Environment.php:280 (Twig\Environment::render)
2 .../simplesamlphp/simplesamlphp/src/SimpleSAML/XHTML/Template.php:560 (SimpleSAML\XHTML\Template::getContents)
1 .../simplesamlphp/simplesamlphp/src/SimpleSAML/XHTML/Template.php:577 (SimpleSAML\XHTML\Template::send)
0 .../simplesamlphp/simplesamlphp/public/module.php:14 (N/A)
Error report with id 0604edf5 generated.
```

**Additional context**
Overriding templates and defining `pageid` to some value seems to be a whacky workaround.

**Comment by tvdijen** [comment]
I think this should have been `pagetitle` ... Would you mind trying that and see if it solves your issue?

**Comment by htto** [comment]
Wow, that was fast. The error is gone, but I'm not quite sure if `pagetitle` is the right one, as it looks more like a title, i.e. a name with spaces and other characters, than a name to compare against. Plus it is potentially translated, thus requiring to provide the login link name translated as well?

```
[..]
simplesamlphp/modules/core/templates/logout-iframe-wrapper.twig:{% set pagetitle = 'Logging out...'|trans %}
simplesamlphp/modules/core/templates/short_sso_interval.twig:{% set pagetitle = 'SimpleSAMLphp'|trans %}
simplesamlphp/modules/core/templates/loginuserpass.twig:{% set pagetitle = 'Enter your username and password'|trans %}
[..]
```

Though, on the other hand you're the expert. not me ;)

**Comment by tvdijen** [comment]
I'll dig into it later tonight!

**Comment by thijskh** [comment]
I've done some archeology. The core:loginpage_links feature existed in the old templates and still exists in the same way in the new templates (producing an UL list of items below the login form).

A new function was added in the very first commit of the base twig template which might be intended to be the same, or something different, in any case uses the same generic name 'links' so setting the template variable 'links' triggers both pieces of code.

I've resolved this by keeping the working function as it was and giving it a better template variable name, and removing the new function added in the base twig which purpose is unclear.

The only thing missing: this is an undocumented feature. But where do we document authsources options?

**Comment by tvdijen** [comment]
I can't think of a better place than https://github.com/simplesamlphp/simplesamlphp/blob/master/docs/simplesamlphp-authsource.md

**Comment by htto** [comment]
@thijskh Thanks for fixing this so quick. Any pointers as to when the next release can be expected/is planned?

**Comment by thijskh** [comment]
We've released 2.0.0 last week. I expect that we will collect the initial experiences with this release for a little while more and then release 2.0.1.
